### PR TITLE
Add DeepBGC output parsing support

### PR DESCRIPTION
**Pull request by prihoda**
Hi guys!

I wanted to use your tool to analyse outputs of our DeepBGC method.

It turns out it was actually pretty simple, since our output mostly follows the output format of antiSMASH (before version 5). 

Would you consider adding these changes to your master branch version?

Here is the usage:
```bash
# RiPP only
python RRE.py -i mibig_example_deepbgc_0.1.8.full.gbk --deepbgc ripp --mode precision deepbgc_ripp        

# Reading in file mibig_example_deepbgc_0.1.8.full.gbk
# Continuing with 150 queries
# Rewriting fasta
# hmmsearch --cpu 1 -o output/deepbgc_ripp/results/RREfam_hmm_results.txt --domtblout output/deepbgc_ripp/results/RREfam_hmm_results.tbl -T 25 data/hmm/RREFam_v4.hmm output/deepbgc_ripp/fastas/fasta_all.fasta
# Finished. Total time: 2.83 seconds (on 1 cores)
# RREFam hits found: 8 out of 150

# All clusters
python RRE.py -i mibig_example_deepbgc_0.1.8.full.gbk --deepbgc clusters --mode precision deepbgc_clusters

# Reading in file mibig_example_deepbgc_0.1.8.full.gbk
# Continuing with 2964 queries
# Rewriting fasta
# hmmsearch --cpu 1 -o output/deepbgc_clusters/results/RREfam_hmm_results.txt --domtblout output/deepbgc_clusters/results/RREfam_hmm_results.tbl -T 25 data/hmm/RREFam_v4.hmm output/deepbgc_clusters/fastas/fasta_all.fasta
# Finished. Total time: 4.64 seconds (on 1 cores)
# RREFam hits found: 8 out of 2964

# All genes
python RRE.py -i mibig_example_deepbgc_0.1.8.full.gbk --deepbgc all --mode precision deepbgc_all          

# Reading in file mibig_example_deepbgc_0.1.8.full.gbk
# Continuing with 3265 queries
# Rewriting fasta
# hmmsearch --cpu 1 -o output/deepbgc_all/results/RREfam_hmm_results.txt --domtblout output/deepbgc_all/results/RREfam_hmm_results.tbl -T 25 data/hmm/RREFam_v4.hmm output/deepbgc_all/fastas/fasta_all.fasta
# Finished. Total time: 4.70 seconds (on 1 cores)
# RREFam hits found: 8 out of 3265
```
Input file:
[mibig_example_deepbgc_0.1.8.full.gbk.gz](https://github.com/Alexamk/RREFinder/files/4403711/mibig_example_deepbgc_0.1.8.full.gbk.gz)

Our publication:
A deep learning genome-mining strategy for biosynthetic gene cluster prediction, Geoffrey D Hannigan,  David Prihoda et al., Nucleic Acids Research, gkz654, https://doi.org/10.1093/nar/gkz654

### CODE REVIEW

File: RRE.py [hunk 1]
```diff
@@ -79,19 +79,34 @@ def find_antismash_clusters(all_seqs,req_type=False):
             if feature.type == 'cluster':
                 coords = int(feature.location.start),int(feature.location.end)
                 quals = feature.qualifiers
-                cluster_type = quals['product'][0]
+                products = quals.get('products', [])
+                product_classes = quals.get('product_class', [])
```
Comment (prihoda) [inline RRE.py:83]: In DeepBGC, the product types are stored in `product_class` qualifier.

File: RRE.py [hunk 2]
```diff
@@ -79,19 +79,32 @@ def find_antismash_clusters(all_seqs,req_type=False):
...
                 if req_type:
                     # Only parse clusters of the required types
                     # Only used for now with the RiPP types
                     cluster_types = cluster_type.split('-')
                     if not any([c in req_type for c in cluster_types]):
                         continue
-                notes = quals['note']
-                cluster_nr = notes[0]
+                notes = quals.get('note')
+                candidate_ids = quals.get('bgc_candidate_id')
```
Comment (prihoda) [inline RRE.py:97]: BGC ID is stored in `bgc_candidate_id` in DeepBGC

**Comment by Alexamk** [comment]
Hello, thanks again for the proposal. Sounds like a good idea to me to integrate it. 
Since I had just updated the same parts to parse antiSMASH version, I've had to adapt your code a bit to merge it well. It works on the test case you sent me, so it should be good to go. Just for the record, for DeepBGC, the gene cluster product is now first determined by the product_class identifier, but if it doesn't find anything there it will use the product identifier. This seemed to make the most sense based on the .gbk file you sent me, although we can adapt it if necessary.

**Comment by prihoda** [comment]
You're welcome, that sounds great. If your publication is still in process, I would be grateful for a citation of our paper.

**Pull request merged**